custom_components/google_home/entity.py

```diff
@@ -183,26 +30,22 @@ def device_info(self) -> DeviceInfo:
         return {
             "identifiers": {(DOMAIN, self.device_name)},
             "name": f"{DEFAULT_NAME} {self.device_name}",
-            "model": VERSION,
             "manufacturer": MANUFACTURER,
         }
 
-    @property
-    def name(self) -> str:
-        """Return the name of the sensor."""
-        return f"{self.device_name} {LABEL_NEXT_TIMER}"
-
-    @property
-    def unique_id(self) -> str:
-        """Return a unique ID to use for this entity."""
-        return f"{self.device_name}/{LABEL_NEXT_TIMER}"
-
-    @property
-    def icon(self) -> str:
-        """Icon to use in the frontend, if any."""
-        return ICON_TIMERS
-
-    @property
-    def device_class(self) -> str:
-        """Return the device class of the sensor."""
-        return DEVICE_CLASS_TIMESTAMP
+    def get_device(self):
+        """Return the device matched by device name
+        from the list of google devices in coordinator_data"""
+        return next(
+            (
+                device
+                for device in self.coordinator.data
+                if device.name == self.device_name
+            ),
+            None,
+        )
```

Comment:
I still think this iterator might be problematic and we would be safer with something simpler like `for .. in ...:` and returning when the device is found. But let's keep it for further investigations and PRs

Reply:
Used normal list here.